test/common.py

```diff
@@ -135,8 +135,21 @@ def assertTensorsSlowEqual(self, x, y, prec=None, message=''):
             max_err = max(max_err, abs(x[index] - y[index]))
         self.assertLessEqual(max_err, prec, message)
 
+    def safeToDense(self, t):
+        r = self.safeCoalesce(t)
+        return r.to_dense()
```

Comment:
why are these `safe` methods needed? In case there are bugs in `coalesce`?

Reply:
Two-fold. One is in case there is a bug in coalesce. But second is, they also check that the structure of a sparse tensor makes sense. For example, `safeToDense` will detect if `t` incorrectly claims itself is coalesced, but it's not (and is sufficient to find the bug #3162).

I would actually prefer it if these were DEBUG enabled assertions in THS, but this is not the "Torch" way.